numpy/lib/_index_tricks_impl.py

```diff
@@ -788,9 +788,12 @@ def _fill_diagonal_dispatcher(a, val, wrap=None):
 def fill_diagonal(a, val, wrap=False):
     """Fill the main diagonal of the given array of any dimensionality.
 
-    For an array `a` with ``a.ndim >= 2``, the diagonal is the list of
-    locations with indices ``a[i, ..., i]`` all identical. This function
-    modifies the input array in-place, it does not return a value.
+    For an array `a` with ``a.ndim >= 2`` when ``wrap == False`` 
+    the diagonal consists of the values with indices 
+    ``[(i, ..., i) for i in range(min(a.shape))]``,
+    otherwise the diagonal consists of values with indices
+    ``[(i % a.shape[0], ..., i % a.shape[-1]) for i in range(max(a.shape))]``.
```

Comment:
IMO these formulae are confusing (and I think incorrect in general). I'd prefer something along the lines of [@mdhaber 's suggestion](https://github.com/numpy/numpy/compare/main...mdhaber:numpy:gh25077) which I think is sufficient to address the ambiguity raised in the original issue but preserves the brevity and clarity of the short description.

Reply:
Ok, I made some changes. But should this PR exist in light of [@mdhaber 's suggestion](https://github.com/numpy/numpy/compare/main...mdhaber:numpy:gh25077), which I did not see before?